### PR TITLE
feat(components): Add json format for BigQuery output to GCS

### PR DESCRIPTION
**Pull request by shikajiro**
Currently, BigQuery output to GCS only supports CSV.
Given the fact that there are repeating columns, we have added support for outputting to json.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

**Comment by Ark-kun** [comment]
Are you planning to expose it in the `component.yaml`?

**Comment by Ark-kun** [comment]
Have you tested this in practice?
/ok-to-test

**Comment by shikajiro** [comment]
@Ark-kun 

Thanks for the review, I forgot to commit component.yaml, so I added it.

I actually have it working with AIPlatform Pipeline.

**Comment by shikajiro** [comment]
Didn't update the test, will add it later!

File: components/gcp/bigquery/query/to_gcs/component.yaml [hunk 1]
```diff
@@ -42,6 +42,10 @@ inputs:
     description: 'The path to the Cloud Storage bucket to store the query output.'
     default: ''
     type: GCSPath
+  - name: output_destination_format
```
Comment (Ark-kun) [inline components/gcp/bigquery/query/to_gcs/component.yaml:45]: This input is not passed to the command-line program.

Comment (shikajiro) [reply components/gcp/bigquery/query/to_gcs/component.yaml:45]: Thanks for the review.
I forgot to add it.

**Comment by shikajiro** [comment]
@Ark-kun 
Added a test to verify that it works locally. Please review.

**Comment by Ark-kun** [comment]
Thank you for the improvement
/lgtm
/approve

**Pull request merged**